### ISSUE TITLE
"Use*BallAboveCP" setting not working

### ISSUE THREAD

**Issue by joya0000** [NECROBOTIO/NecroBot#1974]
Hey guys, I'm have problem with Use*BallAboveCP setting. For example I have "UseGreatBallAboveCP" set as 700, but it is still using it on pokemons below cp 700. Any other settings that I should be aware of?

**Comment by Squirrel3098** [comment]
hello, i believe this problem is linked to the applications new cfg options that say:
 "use  *balltype*  aboveIV" 
I threw a band aide on this by upping my min IV for great balls to 80 and 90 for ultra... i do not know if this has negative effects on catching high CP pokemon that have a low IV. as the app likes to favor IV. Would like some input from the developers.

**Comment by Squirrel3098** [comment]
Ok 
UPDATE: Set min IV for ultra and great balls to 99 or 100. (unless otherwise desired). 
The way their code works is that the CP and IV are checked in an IF/OR saying that if it meets either or both of the requirements to use said ball type then go for it. So what i did was set my IV to 99 because to my understanding IV does not effect a mon's chance of running like CP does. there you go. no more wasted ultra balls.

**Comment by 17Curfis** [comment]
@Squirrel3098 can u explain what this:

  "UseGreatBallBelowCatchProbability": 0.0,
  "UseUltraBallBelowCatchProbability": 0.0,
  "UseMasterBallBelowCatchProbability": 0.0,

and how to setting ? thanks

**Comment by joya0000** [comment]
@Squirrel3098 Thank you man! it is working, which is weird cause I had the setting for minIV as 80 and 90 before, and it wasn't working.

**Comment by Squirrel3098** [comment]
@17Curfis 
So under the hood of PoGo niantics servers generate a random catch probability for pokemon based on CP and IV. 

TL;DR if your looking to just stop the waste of ultra balls but still collect all the pokemon possible, don't worry about this setting. 

I believe what the bot developer was looking at doing for this feature is to allow advanced users to ensure that they catch the top tier pokemon (aka those that have like a 5% or lower catch probability.) for now though i have found no use for it and leaving it default seems to nullify the feature.

**Comment by Squirrel3098** [comment]
@joya0000 yeah, maybe you were just getting lucky with your IV's... either way it seems the latest version is working well and recycling should only get rid of regular pokeballs until it has to remove higher tier ones to keep up with its max allowed pokeballs. 

Feel free to @Squirrel3098 if your having any other issues.

(I'm not a developer of this Bot, although i am a pretty good programmer.)

**Comment by 17Curfis** [comment]
@Squirrel3098 thanks for the explain, that help so much.

**Comment by joya0000** [comment]
@Squirrel3098  the man, closing this now
